apps/internal/base/base.go

```diff
@@ -328,32 +365,33 @@ func (b Client) AcquireTokenByAuthCode(ctx context.Context, authCodeParams Acqui
 
 // AcquireTokenOnBehalfOf acquires a security token for an app using middle tier apps access token.
 func (b Client) AcquireTokenOnBehalfOf(ctx context.Context, onBehalfOfParams AcquireTokenOnBehalfOfParameters) (AuthResult, error) {
-	authParams, err := b.AuthParams.WithTenant(onBehalfOfParams.TenantID)
-	if err != nil {
-		return AuthResult{}, err
-	}
-	authParams.Scopes = onBehalfOfParams.Scopes
-	authParams.AuthorizationType = authority.ATOnBehalfOf
-	authParams.UserAssertion = onBehalfOfParams.UserAssertion
-
+	var ar AuthResult
 	silentParameters := AcquireTokenSilentParameters{
 		Scopes:            onBehalfOfParams.Scopes,
 		RequestType:       accesstokens.ATConfidential,
 		Credential:        onBehalfOfParams.Credential,
 		UserAssertion:     onBehalfOfParams.UserAssertion,
 		AuthorizationType: authority.ATOnBehalfOf,
 		TenantID:          onBehalfOfParams.TenantID,
+		Claims:            onBehalfOfParams.Claims,
 	}
-	token, err := b.AcquireTokenSilent(ctx, silentParameters)
+	ar, err := b.AcquireTokenSilent(ctx, silentParameters)
+	if err == nil {
+		return ar, err
+	}
+	authParams, err := b.AuthParams.WithTenant(onBehalfOfParams.TenantID)
```

Comment:
Previously, this was called before `b.AcquireTokenSilent()`.  Is the change in ordering significant?

Reply:
This is an attempt to clarify the logic. `AcquireTokenSilent()` doesn't use `authParams`, so there's no reason to assign it before the call.